core/systems/disks/steam_removable_media_manager.gd

```
@@ -0,0 +1,241 @@
+	logger.debug("init_capable: " + str(init_capable))
+	logger.debug("format_sd_capable: " + str(format_sd_capable))
+	logger.debug("retrigger_capable: " + str(retrigger_capable))
+	logger.debug("trim_capable: " + str(trim_capable))
+
+	# Get the current drives and ensure they are always up to date
+	udisks2.devices_updated.connect(update_drives)
+	update_drives()
+
+
+## Rebuilds the device_tree with the current device list
+func update_drives(devices: Array[BlockDevice] = []):
+	if devices == []:
+		logger.debug("Got empty array, grabbing devices from udisks2")
+		devices = udisks2.get_devices()
+	self.device_tree = []
+	for block in devices:
+		if not _block_device_has_protected_mount(block):
+			logger.debug(block.dbus_path, "has no protected mounts.")
+			device_tree.append(block)
+	self.drives_updated.emit(self.device_tree)
+
+
+## Calls the SteamRemovableMedia format-media script to format a drive as EXT4
+## and intialize it as a steam library
+func format_drive(device: BlockDevice) -> void:
+	if not format_capable:
+		logger.error("System is not format capable")
+		return
+
+	# This should never hit if using our device tree, but  the method is public so
+	# make sure.
+	if _block_device_has_protected_mount(device):
+		logger.error("Attempted to format device with protected mount. Illegal Operation.")
+		return
+
+	var drive = "/dev" + device.dbus_path.trim_prefix(BLOCK_PREFIX)
+	logger.debug("Formatting drive: " + drive)
+	var args := ["--full", "--device", drive]
+	var result = await _execute_in_thread(SRM_FORMAT_MEDIA, args)
+	logger.debug("Got results: " + str(result[0]), str(result[1]))
+	var success: bool = result[1] == 0
+	device.format_complete.emit(success)
+
+
+## Calls the SteamRemovableMedia init-media script to intialize a drive as a
+## steam library
+func init_steam_lib(partition: PartitionDevice) -> void:
+	if not init_capable:
+		logger.error("System cannot initialize steam libraries")
+		return
```

We should also return a custom error code here.